pinot-connectors/pinot-flink-connector/src/main/java/org/apache/pinot/connector/flink/sink/PinotSinkFunction.java

```diff
@@ -119,7 +139,7 @@ public void invoke(T value, Context context)
       throws Exception {
     _segmentWriter.collect(_recordConverter.convertToRow(value));
     _segmentNumRecord++;
-    if (_segmentNumRecord > _segmentFlushMaxNumRecords) {
+    if (_segmentNumRecord >= _segmentFlushMaxNumRecords) {
```

Comment:
curious, why is this required?

Reply:
This was a bugfix to make flush happen at `_segmentFlushMaxNumRecords` rows and not `_segmentFlushMaxNumRecords + 1`